### ISSUE TITLE
Suggestion: tldr pages for file format man pages

### ISSUE THREAD

**Issue by azraeljd** [tldr-pages/tldr#7744]
Hi! I have a suggestion to create tldr pages for file format (5) man pages, e.g. `deb(5)`.

What do you think?

**Comment by git-em** [comment]
Perhaps would be good to add a dot before the format names, to avoid potential ambiguity (`tldr .deb`).

**Comment by waldyrious** [comment]
Could you share an example of what you envision such a page to contain?

**Comment by azraeljd** [comment]
> Hey @azraeljd, that's an intriguing idea. Could you share an example of what you envision such a page to contain?

Such a page would contain a short description about the file or filename extension, the media type (MIME type) (if exists), and apps that are used to open the file or files with the file extension described in the page.

An example would be `deb`, where a page would look something like this:
```
.deb

The Debian package format.

File name/extension: .deb

Media type: application/vnd.debian.binary-package

Open a .deb package: sudo dpkg --install {{path/to/deb/file}}

Install a .deb package from a remote repository: sudo apt install {{package_name}}
```

**Comment by waldyrious** [comment]
I'm not sure this fits the tldr-pages model. Our focus is examples-based instructions for using CLI tools. In the particular case of `.deb` there are commands that are arguably the canonical way to handle them for the common use cases, but what would we suggest for `.txt`, or `.csv`, or `.jpg`? The things that people will want to do with such files vary wildly, as do the tools they may do those things with. It would be a much more opinionated approach than what we currently have, and I'm neither comfortable with that, nor I am confident that the approach would scale well.

**Comment by azraeljd** [comment]
> I'm not sure this fits the tldr-pages model. Our focus is examples-based instructions for using CLI tools. In the particular case of `.deb` there are commands that are arguably the canonical way to handle them for the common use cases, but what would we suggest for `.txt`, or `.csv`, or `.jpg`?

I'm not sure if there are man pages for `.txt`, `.csv`, or `.jpg`. What I'm referring to in this suggestion are files or filename extensions that has a man page, such as `.deb`, `resolv.conf`, etc.

A man page is more on the technical side of things, and a new user should never touch those things. That's why tldr-pages are here. In my point of view, tldr-pages, while intended for CLI commands, should also be available for all types of man pages.

> The things that people will want to do with such files vary wildly, as do the tools they may do those things with. It would be a much more opinionated approach than what we currently have

The page could just contain the general use case of the file. For example, `.csv`:

```
.csv

Comma-separated values.

File name/extension: .csv

Media type: text/csv

General usage: ...
```

It's so that the reader at least knows what the file is used for.

**Comment by waldyrious** [comment]
I don't dispute the usefulness of that content, but why would we stop there? We could have cheatsheets for regex, SQL, the filesystem hierarchy standard, XDG directories, and a bunch of other topics that are useful... My point is that by staying focused on a specific content type (usage examples for CLI tools), and a specific content format, it's easier to implement linters, clients, translation systems; to decide on and apply content guidelines; to measure coverage against similar documentation projects; and so on. We can't be everything to everyone, as that would spread us thin across many fronts. In good Unix spirit, I say we should do one thing and do it well; other projects can fill up the missing niches :)

**Comment by CleanMachine1** [comment]
Im a little confused here. Is this creating manual pages or tldr pages for file extensions?

**Comment by sbrl** [comment]
Perhaps not for the main tldr repo since the scope here is quite well definted, but it sounds like a great idea for a spin-off project!

**Comment by azraeljd** [comment]
> I don't dispute the usefulness of that content, but why would we stop there? We could have cheatsheets for regex, SQL, the filesystem hierarchy standard, XDG directories, and a bunch of other topics that are useful... My point is that by staying focused on a specific content type (usage examples for CLI tools, whether they have manpages or not), and a specific content format, it's easier to implement linters, clients, translation systems; to decide on and apply content guidelines; to measure coverage against similar documentation projects; and so on. We can't be everything to everyone, as that would spread us thin across many fronts. In good Unix spirit, I say we should do one thing and do it well; other projects can fill up the missing niches :)

Might start a new project myself, thanks for pointing out the point on the Unix philosophy there, tldr-pages does the thing it does very well.

> Im a little confused here. Is this creating manual pages or tldr pages for explaining file extensions?

tldr pages, of course. A man page is very lenghty, and though the concept is similar to the structure of a man page, it is a lot shorter than a man page.

> Perhaps not for the main tldr repo since the scope here is quite well definted, but it sounds like a great idea for a spin-off project!

Thank you. Might start a spin-off project myself.

I will be closing this issue, thank you everyone for your feedback, it is highly appreciated.

**Comment by waldyrious** [comment]
Thank you for raising these questions! It's important for the project to continually revise its vision and ensure we're both sticking to the original goals and being mindful of opportunities to evolve in a way that's compatible with them :)

**Comment by simonmichael** [comment]
For the record, this was something I wanted too as a tldr user and project documenter. Since tldr is such an effective project and tool, I would like to use it (not a separate tool) to look up quick file format docs/cheatsheets, not just command lines. I saw some fears about scope creep, but would relaxing the "commands only" rule, with or even without new scope guidelines, be so bad ? I think tldr would keep on ticking.

**Comment by simonmichael** [comment]
I made a mockup, just to see:

~$ **tldr -r /Users/simon/src/hledger/doc/tldr/hledger-journal.md**

```
hledger journal format

hledger's main data format: a plain text representation of an accounting General Journal.
Files in journal format are often named with
a `.journal` file extension (suggesting a file compatible with hledger and ledger),
or `.hledger` (suggesting a file compatible with hledger only),
or sometimes `.j` for brevity.
More information: <https://hledger.org/hledger.html#journal>.

# This line is commented.
; This line is also commented.

2008/01/01 income             ; <- transaction's first line starts in column 0, contains date and description
    assets:bank:checking  $1  ; <- posting lines start with whitespace, each contains an account name
    income:salary        $-1  ;    followed by at least two spaces and an amount

2008/06/01 gift
    assets:bank:checking  $1  ; <- at least two postings in a transaction
    income:gifts         $-1  ; <- their amounts must balance to 0

2008/06/02 save
    assets:bank:saving    $1
    assets:bank:checking      ; <- one amount may be omitted; here $-1 is inferred

2008/06/03 eat & shop         ; <- description can be anything
    expenses:food         $1
    expenses:supplies     $1  ; <- this transaction debits two expense accounts
    assets:cash               ; <- $-2 inferred

2008/10/01 take a loan
    assets:bank:checking  $1
    liabilities:debts    $-1

2008/12/31 * pay off          ; <- an optional * or ! after the date means "cleared" (or anything you want)
    liabilities:debts     $1
    assets:bank:checking
```

- tldr currently does not allow pages describing file formats, but if it did, here's how it could look.
- This is formatted carefully to hopefully work around tldr clients' rendering limitations.

**Comment by kbdharun** [comment]
Hi @simonmichael (adding the same comment as the one I sent in chatroom here).

Hi, unfortunately, we wouldn't be able to support custom pages for file formats as the structure proposed in your comment would break rendering in a lot of community clients (Something like this would need a major rework). You are free to add pages in the existing format and we do document some non-commands like environment variables, etc currently.

For your tool, you could ship a manpage for the file format separately and reference it on the main man page (that might work better for your use case).
